corehq/form_processor/utils/general.py

```diff
@@ -1,15 +1,25 @@
 from django.conf import settings
 
-from corehq.toggles import USE_SQL_BACKEND
+from corehq.apps.domain.models import Domain
+from corehq.toggles import USE_SQL_BACKEND, NAMESPACE_DOMAIN
 
 
-def should_use_sql_backend(domain):
+def should_use_sql_backend(domain_name):
     if settings.UNIT_TESTING:
         override = getattr(settings, 'TESTS_SHOULD_USE_SQL_BACKEND', None)
         if override is not None:
             return override
 
-    return USE_SQL_BACKEND.enabled(domain)
+    toggle_enabled = USE_SQL_BACKEND.enabled(domain_name)
+    if toggle_enabled:
+        domain = Domain.get_by_name(domain_name)
+        if not domain.use_sql_backend:
+            domain.use_sql_backend = True
+            domain.save()
+            USE_SQL_BACKEND.set(domain_name, enabled=False, namespace=NAMESPACE_DOMAIN)
+        return True
```

Comment:
seems ok to me. the other alternative i could think of would be just deleting the toggle and running a one time migration.